engine.py

```diff
@@ -755,6 +767,35 @@ def show_panel(self, panel_id, title, bundle, widget_class, *args, **kwargs):
         # resolved by looking at the stream of event and force triggering refreshes at the
         # right locations
         #
-        tk_maya.install_callbacks(panel_id, widget_id)
+        tk_maya.install_callbacks(maya_panel_name, widget_id)
 
         return widget_instance
+
+    def close_dialogs_and_panels(self):
+        """
+        Closes the dialogs and panels opened by the engine.
+        """
+
+        # Make a copy of the list of Tank dialogs that have been created by the engine and
+        # are still opened since the original list will be updated when each dialog is closed.
+        opened_dialog_list = self.created_qt_dialogs[:]
+
+        # Loop through the list of opened Tank dialogs.
+        for dialog in opened_dialog_list:
+            # Close the dialog and let its close callback remove it from the original dialog list.
```

Comment:
I understand your point. If ever the dialog is in a shady state and the call to `close()` would crash, this is now covered by a `try/except`.